create the skeleton for the public notebook.

bouzaghrane opened the pull request:
Starting this pull request with creating the skeleton for my public notebook.

timothyb0912 committed:
Initializing files for direct, predictive checks of the deconfounder's assumptions.

timothyb0912 committed:
Initial draft of code to perform the predictive checks.

timothyb0912 committed:
Keeping the environment file updated.

timothyb0912 committed:
Adding data from Wang and Blei (2018)

timothyb0912 committed:
Merge pull request #75 from hassanobeid1994/upload-nmesdata

Adding data from Wang and Blei (2018)

timothyb0912 committed:
Merge branch 'master' into deconfounder-direct-check

timothyb0912 committed:
Adding specification of factor model and utility for drive alone.

timothyb0912 committed:
[WIP] Prior predictive conditional independence tests are functional.

timothyb0912 committed:
Grammar edit.

Commit:
adding csv contining Z and W information from PPCA

timothyb0912 committed:
Completed demo of the predictive conditional independence tests.

timothyb0912 committed:
Moving data files to data folder.

timothyb0912 committed:
Removing data files from the notebook directory.

timothyb0912 committed:
Updating the retro for this week.

timothyb0912 committed:
Merge pull request #81 from hassanobeid1994/update-tb-retro-apr5

Updating the retro for this week.

Commit:
Update mb_retro.md

Commit:
Merge pull request #82 from hassanobeid1994/mb-retro-apr5

Update mb_retro.md

timothyb0912 committed:
Update tb_retro.md

Small typo fix.

hassan-obeid committed:
Update ho_retro.md

hassan-obeid committed:
Update ho_retro.md

timothyb0912 committed:
Responding to comments.

timothyb0912 committed:
Merge pull request #67 from hassanobeid1994/deconfounder-direct-check

Predictive checks of deconfounder assumptions

timothyb0912 committed:
Initial edit to vision doc.

timothyb0912 committed:
Removing hyperbole.

timothyb0912 committed:
Making vission more specific to datasets uesd and the precise methods of demonstration in each part of the project.

timothyb0912 committed:
Editing presentation requirements.

timothyb0912 committed:
Adding time estimates for presentation.

timothyb0912 committed:
Beginning to add non-goals.

timothyb0912 committed:
Editing spacing for readability.

timothyb0912 committed:
Making sure there is a section for code requirements.

timothyb0912 committed:
High-level outline of code requirements.

timothyb0912 committed:
Enumeration of non-goals.

timothyb0912 committed:
Medium-level specificity of code requirements.

timothyb0912 committed:
Removing redundant info and clarifying goals of public notebooks.

timothyb0912 committed:
Adding expected plots from deconfounder investigation.

timothyb0912 committed:
Adding retro for week preceding April 12th, 2020.

timothyb0912 committed:
Merge pull request #90 from hassanobeid1994/update-tb-retro-Apr12

Adding retro for week preceding April 12th, 2020.

Commit:
Update mb_retro.md

Commit:
Merge pull request #91 from hassanobeid1994/update-mb-retro-apr12

Update mb_retro.md

hassan-obeid committed:
Update ho_retro.md

timothyb0912 committed:
Moving refactoring related documents to common folder.

timothyb0912 committed:
Things to look out for in one's code and tests to be added.
All lists are non-exhaustive.

timothyb0912 committed:
Moved refactoring_guide to refactoring folder.

timothyb0912 committed:
Adding programmatic test to notebook to enable refactoring.

Commit:
Update mb_retro.md

Commit:
Merge pull request #96 from hassanobeid1994/update-mb-retro-apr19

Update mb_retro.md

hassan-obeid committed:
Update ho_retro.md

timothyb0912 committed:
Adding causal discovery toolbox to environment file.

Commit:
Update mb_retro.md

Commit:
Merge pull request #105 from hassanobeid1994/bouzaghrane-retro-apr26

Update mb_retro.md

timothyb0912 committed:
Merge pull request #85 from hassanobeid1994/update-vision-and-reqs

Vision & Requirement Updates

hassan-obeid committed:
Update ho_retro.md

timothyb0912 committed:
Updating retro for May 10th, and putting one line per sentence.

timothyb0912 committed:
Merge pull request #106 from hassanobeid1994/update-tb-retro-May10

Updating retro for May 10th, and putting one line per sentence.

Commit:
Update mb_retro.md

Commit:
Merge pull request #107 from hassanobeid1994/mo_retro_may10

Update mb_retro.md

timothyb0912 committed:
Fixing markdown formatting.

timothyb0912 committed:
More formatting fixes.

timothyb0912 committed:
Merge pull request #108 from hassanobeid1994/update-tb-retro-May10

Update tb retro may10--formatting

timothyb0912 committed:
Creating separate files for conditional independence testing functions.

timothyb0912 committed:
Fixing  matplotlib user warnings.

timothyb0912 committed:
Moving testing functions to separate file, changing notebook description, and removing unnecessary cells.

timothyb0912 committed:
Removed internal filtering of dataframe and arrays. Added argument validation.

timothyb0912 committed:
Moved array filtering to notebook instead of within internal functions.

timothyb0912 committed:
Changes to allow visual_permutation_test to perform both marginal and conditional independence tests.

timothyb0912 committed:
Updating marginal independence notebooks to use refactored functions.

timothyb0912 committed:
Adding potential example of potential usefulness of causal graphs in context of recent paper.

timothyb0912 committed:
Adding paper to repo.

timothyb0912 committed:
Updating descriptions to relate to our work.

timothyb0912 committed:
Moving causal graph definition out of notebook into a persisting src directory.

timothyb0912 committed:
Moving causal graph definition to src directory and adding description of latent confounding issues.

timothyb0912 committed:
Adding mpypy to environment specification.

timothyb0912 committed:
Adding mypy type annotations.

timothyb0912 committed:
Adding docstrings for falsification tests of observable independence assumptions.

timothyb0912 committed:
Renaming conditional_independence.py to observable_independence.py

timothyb0912 committed:
Updating the retro for week preceding May 17th.

timothyb0912 committed:
Fixing formatting.

timothyb0912 committed:
Merge pull request #110 from hassanobeid1994/update-tb-retro-may16

Updating the retro for week preceding May 17th.

Commit:
Update mb_retro.md

timothyb0912 committed:
Adding ability to save plots.

timothyb0912 committed:
Adding utilities of potentially wide usefulness.

timothyb0912 committed:
Adding graph for conditional independence testing example.

timothyb0912 committed:
Making sure graphs is a sub-package.

timothyb0912 committed:
Separating independence testing for workflow from notebook.

timothyb0912 committed:
Adding results of workflow/obs_testing_images.py

timothyb0912 committed:
Refactoring root folder path access.

timothyb0912 committed:
Adding initial refactoring notes.

Commit:
Update mb_retro.md

Commit:
Merge pull request #111 from hassanobeid1994/mb_retro_may17

Update mb_retro.md

timothyb0912 committed:
Beginning file for testing of latent conditional independence assumptions.

timothyb0912 committed:
Updating retrospective for May 24th

timothyb0912 committed:
Updating retro to keep last week's entry.

timothyb0912 committed:
Merge branch 'master' into update-tb-retro-may24

timothyb0912 committed:
Merge pull request #115 from hassanobeid1994/update-tb-retro-may24

Updating retrospective for May 24th

timothyb0912 committed:
Put drive alone utility graph in separate file.

timothyb0912 committed:
Added visual_predictive_cit and regression-style simulated cdf plotting functions.

timothyb0912 committed:
Refactored notebooks to separate demonstration from underlying code/logic.

timothyb0912 committed:
Organizing notebook code ahead of additional comments and refactoring.

timothyb0912 committed:
[WIP] First draft of factor model sampling function.

timothyb0912 committed:
Adding docstring for create_conditional_independence_testing_results.

timothyb0912 committed:
Fixing type hints, docstring, and function logic.

timothyb0912 committed:
Adding plot styling.

timothyb0912 committed:
Finishing initial refactoring.

timothyb0912 committed:
Updating graph to have vehicle type cause parking fines instead of directly causing parking costs.

timothyb0912 committed:
Updating tb-retro for May 31st

timothyb0912 committed:
Merge pull request #117 from hassanobeid1994/update-tb-retro-may31

Updating tb-retro for May 31st

hassan-obeid committed:
Update ho_retro.md

timothyb0912 committed:
Merge pull request #109 from hassanobeid1994/dcm-paper-example

Example of potential usefulness of causal graphs in context of recent paper.

timothyb0912 committed:
Merge pull request #116 from hassanobeid1994/refactor-latent-cit

Refactor latent cit

timothyb0912 committed:
Merge pull request #95 from hassanobeid1994/tb-refactoring

Tb refactoring

timothyb0912 committed:
Adding public notebook skeleton.

timothyb0912 committed:
Adding space for description of the main idea behind the testing approaches.

Commit:
create the skeleton for the public notebook.

timothyb0912 committed:
Updating retrospective for June 7th.

timothyb0912 committed:
Merge pull request #123 from hassanobeid1994/update-tb-retro-Jun7

Updating retrospective for June 7th.

hassan-obeid committed:
Update ho_retro.md

timothyb0912 committed:
[WIP] Draft of first 4 cells of testing notebook.

timothyb0912 committed:
[WIP] Filled in marginal independence section. Started filling conditional independence section.

timothyb0912 committed:
Making printed message in visualize_permutation_results generic.

timothyb0912 committed:
[WIP] Adding to-do list, main and caveat descriptions for conditional independence tests, and code for latent conditional independence tests.

timothyb0912 committed:
Fixing markdown to be 1 sentence per line.

timothyb0912 committed:
Adding causal graph for latent confounder drive alone utility graph.

timothyb0912 committed:
[WIP] Adding Purpose, Summary, and latent CIT graph and caveats and pitfalls.

timothyb0912 committed:
Pairing with markdown and adding main description for latent CIT.

timothyb0912 committed:
Adding this week's retro and fixing style/grammar of previous entries.

timothyb0912 committed:
Merge pull request #125 from hassanobeid1994/update-retro-jun14

Adding this week's retro and fixing style/grammar of previous entries.

timothyb0912 committed:
Adding standard hypothesis-testing language and increasing written statement clarity.

timothyb0912 committed:
Addressing comments from weekly meeting.

timothyb0912 committed:
Taking both mypy and cdt and bring up to date with master.

timothyb0912 committed:
[WIP] Conducted all tests for Step 1 and 2 of original pc-algorithm.

timothyb0912 committed:
Fixing typo in progressbar description.

timothyb0912 committed:
Adding final computational steps of PC algorithm and results / findings.

timothyb0912 committed:
Adding diagrams of the causal graph at each step of the PC algorithm.

timothyb0912 committed:
Removing unused cdt package.

timothyb0912 committed:
[WIP] Fixing typo with causal graph, fixing mis-placed description, and adding linked reference.

Commit:
Filling out the public notebook.

bouzaghrane commented:
@timothyb0912 Any advice on how to reorganize code within the simulation loop? It is way too long and I think it would be better if we can break it down to separate components outside of the loop for easier editing by users if need be? What's the smoothest way of doing this?

Commit:
Fixed code in the following ways:

- refer to local libraries
- remove unnecesary imports
- fix typos
- added function to simulation.py

timothyb0912 commented:
@bouzaghrane Here are my first comments off an incomplete look at your block of simulation code. I think you should define the major collections of steps that are being performed. I'm thinking over-arching steps like "simulate new data", "estimate new models", "perturb variables", "store probabilities." I would then collect the logic for these steps in classes (try to identify relevant design patterns and use them here). In the for loop, I would essentially just create and use these objects via a single overall function that performs the desired task. I would likely just overwrite / populate the __call__ method.

timothyb0912 committed:
Updating this week's retro.

timothyb0912 committed:
Merge pull request #127 from hassanobeid1994/update-tb-june21

[TB] Updating June 21 Retro

timothyb0912 committed:
Merge pull request #126 from hassanobeid1994/causal-discovery

PC Algorithm investigation

timothyb0912 committed:
Merge pull request #121 from hassanobeid1994/public-testing-notebooks

Public notebooks for causal graph testing

timothyb0912 committed:
Adding retro for June 28th.

timothyb0912 committed:
Fixing semantic error in markdown syntax.

timothyb0912 committed:
Merge pull request #128 from hassanobeid1994/update-tb-june28

Adding retro for June 28th.

bouzaghrane committed:
The following are in this commit:

Sync the notebook to a jupytext
Separate parameters from code
extract computational code to different .py files
extract graphs to separate .py file

bouzaghrane committed:
adding intro/purpose to notebook.

bouzaghrane committed:
More text throughout the notebook.

timothyb0912 committed:
Setting up pre-commit (#130)

* Setting up pre-commit

* Adding black altered final notebooks

* Adding black altered src/viz files

* Adding black altered causal graphs

* Adding black altered miscellaneous src files

* Adding black altered workflow files

* Adding black altered default files

* Adding black altered parking notebook

* Replacing isort with reorder-python-imports

* Adding black altered causal discovery notebooks

* Keeping environment file updated

* Adding black altered CIT notebook

* Adding black altered MIT notebook

* Adding black altered predictive CIT notebooks

timothyb0912 committed:
Set up repo for collaboration on handbook chapter (#132)

* Initializing article directory

* Updating gitignore with latex files

* Testing setup with example text

* Adding bibliography test

* CLI for article compilation

* Adding compilation instructions

timothyb0912 commented on notebooks/final/7.0-mab-selection-on-observables-final.md:
```diff
@@ -0,0 +1,685 @@
+## Purpose 
+
+
+The purpose of this notebook is to illustrate an example of the workflow outlined in [Brathwaite and Walker (2017)](https://arxiv.org/abs/1706.07502). This simple application aims at highlighting the importance of causal structure in estimating causal effects of interest reflecting changes resulting from policy proposals. The basic idea is to show that when we control for intermediate variables o ...
```
nit: Can you put each sentence on a separate line? It makes git diffs more meaningful as it'll allow highlighting just the part of the sentence that changed instead of the whole line containing unchanged sentences.

I think that the purpose of the notebook is to show if one ignores the structure of one's causal graph (i.e., one's treatment assignment mechanism), one can compute heavily biased treatment effect estimates. We contrast treatment effect estimation while using a correct causal graph vs using the naive, all-variables-are-independent default causal graph.

timothyb0912 commented on notebooks/final/7.0-mab-selection-on-observables-final.md:
```diff
@@ -0,0 +1,685 @@
+
+The purpose of this notebook is to illustrate an example of the workflow outlined in [Brathwaite and Walker (2017)](https://arxiv.org/abs/1706.07502). This simple application aims at highlighting the importance of causal structure in estimating causal effects of interest reflecting changes resulting from policy proposals. The basic idea is to show that when we control for intermediate variables o ...
+
+This notebook achieves this objective by using the dataset and the MNL utility specification from [Brathwaite and Walker (2016)](https://arxiv.org/abs/1606.05900).
```
```suggestion
This notebook uses the dataset and the MNL utility specification from [Brathwaite and Walker (2016)](https://arxiv.org/abs/1606.05900) for demonstration.
```

timothyb0912 commented on notebooks/final/7.0-mab-selection-on-observables-final.md:
```diff
@@ -0,0 +1,685 @@
+
+This notebook achieves this objective by using the dataset and the MNL utility specification from [Brathwaite and Walker (2016)](https://arxiv.org/abs/1606.05900).
+The rest of the notebook is organized as follows:
+ - Defining different causal graphs representing different views for how individuals make mode choices. These causal graphs are based on the MNL utility function from [Brathwaite and Walker (2016)](https://arxiv.org/abs/1606.05900)
```
```suggestion
 - Defining different causal graphs representing different views for how individuals make mode choices. These causal graphs are based on the MNL utility functions from [Brathwaite and Walker (2016)](https://arxiv.org/abs/1606.05900)
```

timothyb0912 commented on notebooks/final/7.0-mab-selection-on-observables-final.md:
```diff
@@ -0,0 +1,685 @@
+The rest of the notebook is organized as follows:
+ - Defining different causal graphs representing different views for how individuals make mode choices. These causal graphs are based on the MNL utility function from [Brathwaite and Walker (2016)](https://arxiv.org/abs/1606.05900)
+ - Simulating data based on the different beliefs about the data generating process illustrated by both causal graphs.
+ - Perturbing one of the variables (e.g.: Travel Distance) to imitate scenarios where a policy proposal would affect one or more of the variables represented by nodes in the causal graph.
```
```suggestion
 - Perturbing one of the variables (e.g.: Travel Distance) to simulate a policy intervention.
```

timothyb0912 commented on notebooks/final/7.0-mab-selection-on-observables-final.md:
```diff
@@ -0,0 +1,685 @@
+ - Defining different causal graphs representing different views for how individuals make mode choices. These causal graphs are based on the MNL utility function from [Brathwaite and Walker (2016)](https://arxiv.org/abs/1606.05900)
+ - Simulating data based on the different beliefs about the data generating process illustrated by both causal graphs.
+ - Perturbing one of the variables (e.g.: Travel Distance) to imitate scenarios where a policy proposal would affect one or more of the variables represented by nodes in the causal graph.
+ - Estimating choice models resulting from the different causal graphs and plotting the distribution of average causal effects of the variable perturbation from the previous step
```
```suggestion
 - Calculating and plotting the distributions of treatment effects according to different causal graphs. 
```

timothyb0912 reviewed (changes requested):
I'm not done reviewing yet! I'm just putting a placeholder for the moment.

Commit:
Mb roughest draft (#136)

* Motivating the importance of graphs

* Adding more content.

* additional writing to section 2.

* added specification of MNL from asymmetric paper.

* remaining of section 2 and section 8.

* section 7.

* added images.

timothyb0912 committed:
Revert "Set up repo for collaboration on handbook chapter (#132)"

This reverts commit d7ca3c08e2fb13fe7eafe67564643c0a546f5240.

timothyb0912 committed:
Merge branch 'master' into mb_public_notebooks

timothyb0912 committed:
Adding jupytext pre-commit

timothyb0912 committed:
Restoring Mo's state

timothyb0912 committed:
Restoring Mo's final notebook

timothyb0912 committed:
Alphabetizing packages

timothyb0912 committed:
Add fitter to environment.yaml

timothyb0912 committed:
Moving fitter to pip install

timothyb0912 committed:
Merge branch 'mb_public_notebooks' of https://github.com/hassan-obeid/tr_b_causal_2020 into mb_public_notebooks

timothyb0912 committed:
Restoring Mo's observables module

timothyb0912 committed:
Trying again to restore Mo's observable's module

timothyb0912 commented on src/observables/distfit.py:
```diff
@@ -123,21 +128,20 @@ def get_continuous_dist(var, var_val, cont_dists, alt_name=None):
-                           timeout=60,
-                           verbose=True)
+    fitter_object = Fitter(
+        data=var_val, distributions=cont_dists, timeout=60, verbose=True
```
The `verbose=True` specification gives me the following error when I try to run your public notebook on my computer.
<img width="1140" alt="Screen Shot 2020-11-07 at 3 31 38 PM" src="https://user-images.githubusercontent.com/5533699/98453558-5fc67600-210f-11eb-8eae-85ec5f7aca21.png">

I remember you said you had to do something to fitter to get this to work? Is there anyway we can do without choices like that? It makes it impossible for our work to be reproduced.

bouzaghrane replied:
Just fixed it in the new version. It seems like a newer version of the fitter package gets rid of this verbose argument.

bouzaghrane committed:
fixed error with fitter

The newly updated version of the fitter package gets rid of the verbose argument so I updated the code accordingly. I ran the final notebooks and they ran with no problems. I am yet to run @hassan-obeid notebooks because I am unclear on which ones will be final.

@timothyb0912  any thoughts on how to prevent this type of issue in the future since we rely on third party libraries?
I also changed the version number for black in the environment file since pip wasn't able to find a version 19.10

Commit:
Update notebooks/final/7.0-mab-selection-on-observables-final.md

Co-authored-by: Timothy Brathwaite <timothyb0912@users.noreply.github.com>

Commit:
Update notebooks/final/7.0-mab-selection-on-observables-final.md

Co-authored-by: Timothy Brathwaite <timothyb0912@users.noreply.github.com>

Commit:
accepting suggestions from Tim

Co-authored-by: Timothy Brathwaite <timothyb0912@users.noreply.github.com>

Commit:
accepting suggestions from Tim

Co-authored-by: Timothy Brathwaite <timothyb0912@users.noreply.github.com>

timothyb0912 commented:
Let's merge this if there are no more **immediate** changes you wanna make!

timothyb0912 commented:
🔒 Wait actually, before you do any merging...

Can we pare down the number of affected files in this merge? You're just adding your public notebook and the files in `src` that support it right?

Can you checkout the version of all other files from master and only retain changes to the smallest possible number of files?

timothyb0912 reviewed (changes requested):
Please pare down changes to a minimal set of files.

bouzaghrane commented:
@timothyb0912 okay I'm not entirely sure how to merge just specific files and didn't want to mess this up. I know which files need to be merged and we can talk about merging them tomorrow.

timothyb0912 commented:
@bouzaghrane I take it this can be closed right?

bouzaghrane commented:
Yup! @timothyb0912 closing this.
